capstone/capapi/permissions.py

```diff
@@ -18,9 +19,41 @@ def has_permission(self, request, view):
             # boot if user tries anything but `GET`, `OPTIONS`, or `HEAD`
             return False
 
-        # give away metadata to everyone!
-        if not request.query_params.get('type') == 'download':
+        # check whether user wants to get full case text
```

Comment:
The reason I changed this to a more substantive user permissions check was because a user is an API user whether or not they have the API token. I want to provide as much access to the API as possible, and that means checking whether a case is whitelisted and if it is, allowing a user through even when they are anonymous. Maybe there should be a new name for the authenticated user permissions check, like IsAuthenticatedUser?

Reply:
That makes sense! The part I'm getting at here is that this permissions class should *only* be used for the case endpoint -- that's the only endpoint where all the logic here applies, right?

You're also using this class for the resend_verification endpoint, which doesn't work. So it makes sense to change a couple of things:

- Name this something so it should clearly only be used from the case endpoint -- `IsAPIUser` sounds too generic for what it is.
- Use a different class for `resend_verification`.